topk-js/src/data/scalar.rs

```diff
@@ -0,0 +1,95 @@
+use napi::{bindgen_prelude::*, sys::napi_typeof};
+
+use super::utils::is_napi_integer;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum Scalar {
+    Bool(bool),
+    U32(u32),
+    U64(u64),
+    I32(i32),
+    I64(i64),
+    F32(f32),
+    F64(f64),
+    String(String),
+}
+
+impl Into<topk_rs::data::Scalar> for Scalar {
+    fn into(self) -> topk_rs::data::Scalar {
+        match self {
+            Scalar::Bool(b) => topk_rs::data::Scalar::Bool(b),
+            Scalar::U32(u) => topk_rs::data::Scalar::U32(u),
+            Scalar::U64(u) => topk_rs::data::Scalar::U64(u),
+            Scalar::I32(i) => topk_rs::data::Scalar::I32(i),
+            Scalar::I64(i) => topk_rs::data::Scalar::I64(i),
+            Scalar::F32(f) => topk_rs::data::Scalar::F32(f),
+            Scalar::F64(f) => topk_rs::data::Scalar::F64(f),
+            Scalar::String(s) => topk_rs::data::Scalar::String(s),
+        }
+    }
+}
+
+impl Into<topk_protos::v1::data::Value> for Scalar {
+    fn into(self) -> topk_protos::v1::data::Value {
+        match self {
+            Scalar::Bool(b) => topk_protos::v1::data::Value::bool(b),
+            Scalar::U32(u) => topk_protos::v1::data::Value::u32(u),
+            Scalar::U64(u) => topk_protos::v1::data::Value::u64(u),
+            Scalar::I32(i) => topk_protos::v1::data::Value::i32(i),
+            Scalar::I64(i) => topk_protos::v1::data::Value::i64(i),
+            Scalar::F32(f) => topk_protos::v1::data::Value::f32(f),
+            Scalar::F64(f) => topk_protos::v1::data::Value::f64(f),
+            Scalar::String(s) => topk_protos::v1::data::Value::string(s),
+        }
+    }
+}
+
+impl ToNapiValue for Scalar {
+    unsafe fn to_napi_value(
+        env: napi::sys::napi_env,
+        val: Self,
+    ) -> napi::Result<napi::sys::napi_value> {
+        match val {
+            Scalar::Bool(b) => bool::to_napi_value(env, b),
+            Scalar::U32(u) => u32::to_napi_value(env, u),
+            Scalar::U64(u) => u32::to_napi_value(env, u as u32),
```

Comment:
worth adding a TODO here, `u as u32` is lossy so we might consider returning `BigInt` in the future